lib/tasks/s3_file_conversion.rake

```diff
@@ -1,47 +1,113 @@
+          file_name = File.basename(image_url)
+          file_path = File.join(dir_name, file_name)
+
+          # Download the image from S3 and store in temp folder.
+          # * open the tmp file
+          # * fetch the image from s3
+          # * write the content to the tmp file
+          puts "*** START WRITING ***"
+          open(file_path, 'wb') do |file|
+            file.write open(image_url).read
+          end
+
+          # Save the remote image to the model's uploader column
+          # On save the image will overwrite the remote file with the same name
+          puts "*** UPDATE RESOURCE ***"
+          resource.vendor = voice_base
+          resource.image = Rails.root.join('tmp', 's3_uploads', file_name).open
+
+          if resource.save!
+            puts "*** SAVING ***"
+            success << resource
+          else
+            puts "*** FAILED ***"
+            errors << resource
+          end
         end
       end
+    end
+    print_result(success, errors)
+  end
+
+  task audio_conversion: :environment do |task|
+    success = []
+    errors = []
+    voice_base = Vendor.find_by uid: "voice_base"
+
+    Transcript.find_each do |transcript|
+      # CarrierWave introduces an image_url which overrides the call to the
+      # model attribute so we need to use read_attribute to access it
+      audio_url = transcript.read_attribute(:audio_url)
+      puts "*** AUDIO URL: #{audio_url}"
+
+      if audio_url.present? && transcript.audio.blank?
+        puts "*** CONVERT AUDIO ***"
+        # Setup to store file from S3 bucket
+        dir_name = FileUtils.mkpath('tmp/s3_uploads/')
+        file_name = File.basename(audio_url)
+        file_path = File.join(dir_name, file_name)
+
+        # Download the audio files from S3 and store in temp folder.
+        # * open the tmp file
+        # * fetch the audio file from s3
+        # * write the content to the tmp file
+        puts "*** START WRITING ***"
+        open(file_path, 'wb') do |file|
+          file.write open(audio_url).read
+        end
+
+        # Save the remote audio to the model's uploader column
+        # when save the audio will be overwrite the remote file with the same name
+        puts "*** UPDATE TRANSCRIPT ***"
+        transcript.vendor = voice_base
+        transcript.audio = Rails.root.join('tmp', 's3_uploads', file_name).open
+
+        if transcript.save!
+          puts "*** SAVING ***"
+          success << transcript
+        else
+          puts "*** FAILED ***"
+          errors << transcript
+        end
+      end
+    end
+    print_result(success, errors)
+  end
 
-      if success.empty? && errors.empty?
-        puts "No records needed to be updated"
-      else
-        puts "Updated #{success.count} records"
-        puts "Failed to update the #{errors.count} records"
-        errors.each { |el| puts "* #{el.class} - #{el.uid}" }
+  def print_result(success, errors)
+    if success.empty? && errors.empty?
```

Comment:
update the print results method to write the output to a log file (ideally to a new one). E.g `log/carrierwave_uoloader.log`. that way it will be easy for you to read for any errors